### PR TITLE
[REVIEW] Fix dask groupby mutliindex test case issues in join

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
This has fix for two sets of test case failure.
Thank you @rjzamora for helping me on this bug.
Hopefully the last.

### CODE REVIEW

**Comment by rgsl888prabhu** [comment]
Just created couple of more commits to check if anything else fails.

File: python/dask_cudf/dask_cudf/tests/test_groupby.py [hunk 1]
```diff
@@ -321,6 +321,6 @@ def test_groupby_reset_index_drop_True():
     pddf = dd.from_pandas(df.to_pandas(), 5)
     gr = ddf.groupby(["a"]).agg({"b": ["count"]}).reset_index(drop=True)
     pr = pddf.groupby(["a"]).agg({"b": ["count"]}).reset_index(drop=True)
-    gf = gr.compute()
-    pf = pr.compute()
+    gf = gr.compute().sort_values(by=["b"]).reset_index(drop=True)
```
Comment (kkraus14) [inline python/dask_cudf/dask_cudf/tests/test_groupby.py]: Are we dropping the index because we don't currently retain the index on groupbys for cuDF?

Comment (rgsl888prabhu) [reply python/dask_cudf/dask_cudf/tests/test_groupby.py]: No, this is for index drop test and `sort_values(by=["b"])` will add an index/column as far as I remember

Comment (kkraus14) [reply python/dask_cudf/dask_cudf/tests/test_groupby.py]: Oh woops, completely missed the test name, thanks!

**Pull request merged**